torch/overrides.py

```diff
@@ -1852,30 +1842,55 @@ def __torch_function__(self, func, types, args=(), kwargs=None):
 
     def __enter__(self):
         old = _get_torch_function_mode()
-        if hasattr(self, "inner"):
-            raise RuntimeError(f"{self} has already been used as a mode. Please use a fresh version or use restore")
-        else:
-            self.inner = old
-            if old is None:
-                self.ancestors = set()
-            else:
-                self.ancestors = self.inner.ancestors.union({self.inner})
-        _set_torch_function_mode(self)
+        if self in _cur_torch_function_mode or self is old:
```

Comment:
This test is quadratic in the number of pushes you do; while it's a nice test I don't think we should do it this way. If we really want the test, we could maintain state on the mode object saying if it's on a stack or not and query that

Reply:
Hmm okay yeah I'll remove it. I would add it to the state but if we do want the mode stack to be thread local (which Richard brought up on the previous version and I haven't looked into), that wouldn't work with this

Reply:
yeah ok, let's just remove it

Reply:
Removed and changed the test to show the behavior when pushing the same mode twice (mode just runs twice)